src/gantry_calibration/calibrate.py

```diff
@@ -0,0 +1,227 @@
+TRAVEL_RANGE = TRAVEL_MAX - TRAVEL_MIN
+ERROR_FRACTION = 0.05
+
+
+def main() -> None:
+    """Dispatch command for various interpolation methods."""
+    print("*** Gantry Calibration Algorithm ***")
+
+    arguments = sys.argv[1:]
+    args = docopt(__doc__, argv=arguments)
+
+    if args["basic"]:
+        basic_calibration()
+    if args["bilinear"]:
+        if args["<points>"]:
+            points = int(args["<points>"])
+        else:
+            points = 2
+        polynomial_interpolation(1, 1, points, points)
+    elif args["biquadratic"]:
+        if args["<points>"]:
+            points = int(args["<points>"])
+        else:
+            points = 3
+        polynomial_interpolation(2, 2, points, points)
+    elif args["bicubic"]:
+        if args["<points>"]:
+            points = int(args["<points>"])
+        else:
+            points = 4
+        polynomial_interpolation(3, 3, points, points)
+    elif args["poly"]:
+        polynomial_interpolation(
+            int(args["<x_order>"]),
+            int(args["<y_order>"]),
+            int(args["<x_points>"]),
+            int(args["<y_points>"]),
+        )
+
+
+def basic_calibration() -> None:
+    """Demonstrate basic bilinear calibration with annotation."""
+    x_travel = Travel(TRAVEL_MIN, TRAVEL_MAX)
+    y_travel = Travel(TRAVEL_MIN, TRAVEL_MAX)
+    points = generate_points(x_travel, y_travel, 2, 2)
+    calibration = Calibration(1, 1, points)
+    plot(points, calibration.map, 5, annotation=True)
+
+
+def polynomial_interpolation(x_order: int, y_order: int, x_points: int, y_points: int) -> None:
+    """
+    Interpolate with any order of polynomial.
+
+    :param x_order: The order of polynomial to fit the x-axis
+    :param y_order: The order of polynomial to fit the y-axis
+    :param x_points: The number of points along the x-axis used for fitting
+    :param y_points: The number of points along the y-axis used for fitting
+
+    """
+    print(f"Order {x_order} x {y_order} interpolation using {x_points} x {y_points} points")
+    x_travel = Travel(TRAVEL_MIN, TRAVEL_MAX)
+    y_travel = Travel(TRAVEL_MIN, TRAVEL_MAX)
+    points = generate_points(x_travel, y_travel, x_points, y_points)
+    calibration = Calibration(x_order, y_order, points)
+    plot(points, calibration.map, 5)
+
+
+def generate_points(
+    x_travel: Travel, y_travel: Travel, x_count: int, y_count: int
+) -> list[list[PointPair]]:
+    """
+    Generate random points for calibration algorithm.
+
+    :param x_travel: Travel range of points on the x-axis
+    :param y_travel: Travel range of points on the y-axis
+    :param x_count: Number of points to generate on the x-axis
+    :param y_count: Number of points to generate on the y-axis
+    """
+
+    def rand() -> float:
+        """Generate random numbers within range defined by ERROR_FRACTION of TRAVEL_RANGE."""
+        rng = np.random.default_rng()
+        return rng.uniform(-ERROR_FRACTION * TRAVEL_RANGE, ERROR_FRACTION * TRAVEL_RANGE)
```

Comment:
This should not reference the TRAVEL_RANGE global variable. It should be computed internally in the function. It will glitch out if people try to change the X and Y travels independently.

Reply:
Doh... thanks for catching that!  Refactored to calculate it from x_travel and y_travel, and also refactored generate_points() to accept one more parameter, error_fraction, instead of the function just pulling in a global.